x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -201,6 +201,13 @@ export const getDefaultTab = (
   const installedPolicyTemplatesKspm = pluginStatus?.kspm?.status;
   let preferredDashboard: PosturePolicyTemplate = POSTURE_TYPE_CSPM;
 
+  if (
+    installedPolicyTemplatesCspm === 'unprivileged' &&
+    installedPolicyTemplatesKspm === 'unprivileged'
+  ) {
+    return preferredDashboard;
+  }
+
   if (!cspmStats || !pluginStatus) {
```

Comment:
this check and the fact that there are some race conditions we wanted to fix with it https://github.com/elastic/kibana/pull/173853/files#r1488193444 made the function very hard to reason about. It's risky to make any changes to it without e2e/integration tests, as unit tests test implementation not the business logic right now (check for testing for `undefined`